tests/test_daily_temperature_extrema.py

```diff
@@ -0,0 +1,543 @@
+import datetime
+
+from pytest import fixture, mark, approx
+from sqlalchemy.sql import text
+
+from pycds.util import generic_sesh
+from pycds import Network, Station, History, Variable, Obs, NativeFlag, PCICFlag
+from pycds.weather_anomaly import DailyMaxTemperature, DailyMinTemperature
+
+
+def describe_function_effective__day__for__Tmax():
+    @mark.parametrize('tod, obs_time', [
+        ('morning', '2000-01-01 07:23'),
+        ('afternoon', '2000-01-01 16:18')
+    ])
+    @mark.parametrize('freq, expected_day', [
+        ('1-hourly', {'morning': datetime.datetime(2000, 1, 1), 'afternoon': datetime.datetime(2000, 1, 1)}),
+        ('12-hourly', {'morning': datetime.datetime(2000, 1, 1), 'afternoon': datetime.datetime(2000, 1, 2)})
+    ])
+    def it_returns_the_expected_day_of_observation(mod_empty_database_session, tod, obs_time, freq, expected_day):
+        result = mod_empty_database_session.execute(
+            text('SELECT effective_day_for_Tmax(:obs_time, :freq) AS eday'),
+            {'obs_time': obs_time, 'freq': freq}
+        ).fetchall()
+        assert len(result) == 1
+        assert result[0]['eday'] == expected_day[tod]
+
+
+views = [DailyMaxTemperature, DailyMinTemperature]
+
+@fixture(scope='module')
+def with_views_sesh(mod_empty_database_session):
+    sesh = mod_empty_database_session
+    for view in views:
+        view.create(sesh)
+    yield sesh
+    for view in reversed(views):
+        view.drop(sesh)
+
+
+def refresh_views(sesh):
+    for view in views:
+        view.refresh(sesh)
+
+
+def describe_with_1_network():
```

Comment:
The multi-leveled nesting of fixtures and test functions are not overtly obvious to me as to how they work. Could you add a bit of a description (or link to docs) as to how this works and what it's setting up and doing? I don't quite understand why one needs a nesting of four functions deep.

And why does the query fixture have to be defined multiple times?

I'm happy that tests are being run, but I'd be happier if I understood how :)